### ISSUE TITLE
Style/EmptyLiteral is (possibly) not safe

### ISSUE THREAD

**Issue by deadeyejoe** [rubocop-hq/rubocop#6648]
I've encountered a situation where this cop is possibly not safe.

I have a class with subclasses. The constructor for the parent class accepts a hash, and I want this to be an empty hash in one of the child classes.

```ruby
class Parent
  def initialize(hash)
    @data = hash
  end
end

class Child < Parent
  def initialize(string)
    super Hash.new
    @string = string
  end
end
```

When I autocorrect the child class it gets changed to:

```ruby
class Child < Parent
  def initialize(string)
    super {}
    @string = string
  end
end
```
Which is not equivalent. This results in the argument `string` being passed to the super call (i.e. it behaves as a 'bare' `super` call).  Changing this line to `super({})` causes it to work as expected.

--------

```
$ [bundle exec] rubocop -V
0.62.0 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-linux)
```

**Comment by rrosenblum** [comment]
This likely applies to any method, not just `super`, that is accepting `Hash.new` as an argument.

**Comment by rrosenblum** [comment]
Other methods are a non issue. They were already accounted for. `super` was missed because it is considered its own type rather than a send type.
